### PR TITLE
[BEAM-4494] Migrate recent website changes from beam-site to beam

### PR DESCRIPTION
**Pull request by swegner**
------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by swegner** [comment]
R: @alanmyrvold

**Comment by alanmyrvold** [comment]
And I see Website ("Run Website PreCommit") — FAILURE?

**Comment by swegner** [comment]
Looks like the `.gradle` files were added in https://github.com/apache/beam-site/pull/554 . @yifanzou do you know anything about these? I assume it was a mistake; we can follow-up with a subsequent PR either here or on the `beam-site` repository.

Looking into precommit failure now.

**Comment by swegner** [comment]
Run Website PreCommit

**Comment by swegner** [comment]
I believe pre-commits are failing because current_release is updated from 2.6.0 to 2.7.0, but we haven't published the javadocs/pydocs to the new `release-docs` branch yet. Sent a PR here: https://github.com/apache/beam-site/pull/575

File: website/src/documentation/dsls/sql/create-external-table.md [hunk 1]
```diff
@@ -19,20 +21,20 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-# CREATE TABLE
+# CREATE EXTERNAL TABLE
 
-Beam SQL's `CREATE TABLE` statement registers a virtual table that maps to an
+Beam SQL's `CREATE EXTERNAL TABLE` statement registers a virtual table that maps to an
 [external storage system](https://beam.apache.org/documentation/io/built-in/).
```
Comment (alanmyrvold) [inline website/src/documentation/dsls/sql/create-external-table.md]: Should be {{ site.baseurl }}

File: website/src/_posts/2018-10-03-beam-2.7.0.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+We are happy to present the new 2.7.0 release of Beam. This release includes both improvements and new functionality.
+See the [download page](https://beam.apache.org/get-started/downloads/#270-2018-10-02) for this release.<!--more-->
```
Comment (alanmyrvold) [inline website/src/_posts/2018-10-03-beam-2.7.0.md]: Should be {{ site.baseurl }}

**Comment by alanmyrvold** [comment]
Rename of create-external-table.md file causes issues getting external link https://beam.apache.org/documentation/dsls/sql/create-external-table ... this link should use {{ site.baseurl }} to be testable when it is new

File: website/src/documentation/dsls/sql/create-external-table.md (hunk 1 above)
Comment (swegner) [reply website/src/documentation/dsls/sql/create-external-table.md]: Thanks; I will fix these upstream in `beam-site` and then re-import

**Comment by yifanzou** [comment]
> Looks like the `.gradle` files were added in [apache/beam-site#554](https://github.com/apache/beam-site/pull/554)

Seems like gradle files were automatically added by my IntelliJ. The [apache/beam-site/#579](https://github.com/apache/beam-site/pull/579) remove those binary files.

**Comment by swegner** [comment]
Run Website PreCommit

**Comment by swegner** [comment]
Run Website PreCommit

**Comment by swegner** [comment]
Run Website PreCommit

**Comment by swegner** [comment]
I'm not sure why pre-commits are still failing. The error is that it cannot resolve link `https://beam.apache.org/documentation/dsls/sql/create-external-table/`. But it shouldn't be resolving from `https://beam.apache.org`; the link should point to the locally-served site, right?
/cc @udim @alanmyrvold

**Comment by udim** [comment]
Perhaps it's failing on a leftover file `./generated-content/documentation/dsls/sql/create-table/index.html` that's no longer being generated. Perhaps it's a bug in the new process that it doesn't first clear `build/website/generated-content/`.

**Comment by udim** [comment]
I think `baseurl` in `_config.yml` is the issue, which explains why there was a a separate https://github.com/apache/beam-site/blob/asf-site/_config_test.yml on apache/beam-site.

I ran `https://github.com/apache/beam-site/blob/asf-site/_config_test.yml` on my local machine (without this PR) and went to
http://0.0.0.0:4000/documentation/dsls/sql/statements/create-table/. It redirected me to https://beam.apache.org/documentation/dsls/sql/create-table/.

**Pull request merged**